src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker_Generators.cs

```diff
@@ -46,13 +46,6 @@ private partial class CompilationTracker : ICompilationTracker
                 return (compilationWithGeneratedFiles, generatorInfo.Documents, generatorInfo.Driver);
             }
 
-            if (!this.ProjectState.SourceGenerators.Any())
-            {
-                // We don't have any source generators.  Trivially bail out.
-                var compilationWithGeneratedFiles = compilationWithoutGeneratedFiles;
-                return (compilationWithGeneratedFiles, TextDocumentStates<SourceGeneratedDocumentState>.Empty, generatorInfo.Driver);
-            }
```

Comment:
this code was running on the *host* side no matter what.  we do *not* want that.  that basically loads and caches the SourceGenerators info locally, which means (due to .net fx limitations) that it is effectively fixed.  Note: i intend to get rid of ProjectState.SourceGenerators entirely as an api. But this will be a series of small refactorings.